ci/workflow_summary/github.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A utility for GitHub REST API."""
+
+import requests
+import logging
+
+RETRIES = 3
+BACKOFF = 5
+RETRY_STATUS = (403, 500, 502, 504)
+TIMEOUT = 5
+TIMEOUT_LONG = 20
+
+OWNER = ''
+REPO = ''
+BASE_URL = 'https://api.github.com'
+GITHUB_API_URL = '%s/repos/%s/%s' % (BASE_URL, OWNER, REPO)
+
+
+def set_api_url(owner, repo): 
+  if owner and repo:
+    global OWNER, REPO, GITHUB_API_URL
+    OWNER = owner
+    REPO = repo
+    GITHUB_API_URL = '%s/repos/%s/%s' % (BASE_URL, OWNER, REPO)
```

Comment:
Modifying global state can cause subtle and hard to debug issues. You could potentially create a class to wrap these functions and stores `owner` and `repo`.